front/pages/api/w/[wId]/data_sources/request-access.ts

```diff
@@ -0,0 +1,96 @@
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+import sanitizeHtml from "sanitize-html";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { sendEmail } from "@app/lib/email";
+import { apiError } from "@app/logger/withlogging";
+
+export const PostRequestAccessBodySchema = t.type({
+  email: t.string,
+  emailMessage: t.string,
+  emailRequester: t.string,
+  dataSourceName: t.string,
+});
+
+export type PostRequestAccessBody = t.TypeOf<
+  typeof PostRequestAccessBodySchema
+>;
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  auth: Authenticator
+) {
+  const owner = auth.workspace();
+  const user = auth.user();
+
+  if (!user || !owner) {
```

Comment:
We need to check for `auth.isUser()` to ensure that the user has access to the workspace.